### PR TITLE
TestUtils: read all blocks and sort them by handle before comparison.

### PR DESCRIPTION
**Pull request by JinheLin**
### What problem does this PR solve?

Issue Number: ref #5689

Problem Summary:
Blocks can be unordered when read-thread-pool enabled if there are more than one segments. 

To simplify testing, read all blocks and sort them by handle column before comparison.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by Lloyd-Pottiger** [comment]
/cc @ywqzzy 
maybe you are more familiar with the code.

**Review by lidezhu — APPROVED** [review]
LGTM

File: dbms/src/TestUtils/InputStreamTestUtils.cpp [hunk 1]
```diff
@@ -362,5 +363,112 @@ ::testing::AssertionResult InputStreamVSBlockUnrestrictlyCompareColumns(
...
+            return ::testing::AssertionFailure() << reason;
+        }
+        blocks.emplace_back(std::move(read_block));
+    }
+    stream->readSuffix();
+
+    auto cmp_blk = [](const Block & a, const Block & b) {
+        const auto & col_a = a.has(EXTRA_HANDLE_COLUMN_NAME) ? a.getByName(EXTRA_HANDLE_COLUMN_NAME) : a.getByPosition(0);
+        const auto & col_b = b.has(EXTRA_HANDLE_COLUMN_NAME) ? b.getByName(EXTRA_HANDLE_COLUMN_NAME) : b.getByPosition(0);
+        if (col_a.column->size() == 0 || col_b.column->size() == 0)
```
Comment (ywqzzy) [inline dbms/src/TestUtils/InputStreamTestUtils.cpp:420]:
```suggestion
        if (col_a.column->empty() || col_b.column->empty())
```

File: dbms/src/TestUtils/InputStreamTestUtils.cpp [hunk 2]
```diff
@@ -362,5 +363,112 @@ ::testing::AssertionResult InputStreamVSBlockUnrestrictlyCompareColumns(
...
+            const auto & col_name = colnames[col_idx];
+            // Copy the [start_offset, read_block.rows()) of `expect_block`
+            const auto & expect_full_col = expect_block.getByPosition(col_idx);
+            auto expect_col = expect_full_col.cloneEmpty();
+            auto column_data = expect_col.type->createColumn();
+            column_data->insertRangeFrom(*expect_full_col.column, start_offset, read_block.rows());
+            expect_col.column = std::move(column_data);
+
+            const auto & actual_col = read_block.getByName(col_name);
+            if (auto res = DB::tests::columnEqual(expect_col, actual_col); !res)
```
Comment (ywqzzy) [inline dbms/src/TestUtils/InputStreamTestUtils.cpp:444]:
```suggestion
            if (auto res = columnEqual(expect_col, actual_col); !res)
```

**Review by JaySon-Huang — APPROVED** [review]
lgtm

**Pull request merged**